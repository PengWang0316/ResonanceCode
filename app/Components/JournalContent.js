import React, { Component } from "react";
import PropTypes from "prop-types";

class JournalContent extends Component{

  componentWillMount(){
    this.state = {isShared: this.props.isShared};
    this.state[this.props.newContentKey]=this.props.newContent;
    // console.log("isShared: ",this.state.isShared);
  }

  handleChange(event){
    // let newState={};
    // newState[this.props.newContentKey]=event.target.value;
    this.setState({
      [this.props.newContentKey]: event.target.value
    });
    this.props.handleChangeCallback(this.props.newContentKey, event.target.value);
  }

  handleClose(){
    // console.log("handleclose:",contentKey);
    this.props.handleDeleteContentCallback(this.props.newContentKey);
  }

  handleSharedBoxChange(){
    // console.log("isShared", this.state.isShared);
    let isShared = !this.state.isShared;
    this.setState({isShared: isShared});
    this.props.handleSharedBoxChangeCallback(this.props.newContentKey, isShared);
  }

  render(){
    let contentName = this.props.newContentName.replace("_", " ");
    let contentKey = this.props.newContentKey;
    return(
      <div className="form-group form-div">
        <i onClick={()=>{this.handleClose();}} className="fa fa-window-close closeIconSpan" title={`Remove ${contentName}`} />
        <div className="journalContentTitleDiv">
          <div className="row">
            <div className="col-xs-6 text-left">
              <label htmlFor={contentKey} className="col-form-label text-capitalize">{contentName}</label>
            </div>
            <div className="col-xs-6 text-right">
              <label className="checkbox-inline"><input onChange={()=>{this.handleSharedBoxChange();}} type="checkbox" id="inlineCheckbox1" checked={this.state.isShared} /> Share</label>
            </div>
          </div>

        </div>

        <div>
          <textarea className="form-control" rows="3" type="text" value={this.state[contentKey]} placeholder={`${contentName}...`} id={`${contentKey}`} onChange={(event)=>{this.handleChange(event);}} />
        </div>

      </div>
    );
  }

}
JournalContent.propTypes={
  newContent: PropTypes.string.isRequired,
  newContentName: PropTypes.string.isRequired,
  newContentKey: PropTypes.string.isRequired,
  handleChangeCallback: PropTypes.func.isRequired,
  handleDeleteContentCallback: PropTypes.func.isRequired,
  handleSharedBoxChangeCallback: PropTypes.func.isRequired,
  isShared: PropTypes.bool.isRequired
};
export default JournalContent;
