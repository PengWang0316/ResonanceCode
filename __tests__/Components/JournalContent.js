import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import JournalContent from '../../app/Components/JournalContent';

window.$ = jest.fn(() => ({ tooltip: jest.fn() }));

describe('JournalContent test', () => {
  const defaultProps = {
    newContent: 'newContent',
    newContentName: 'newContentName',
    newContentKey: 'newContentKey',
    handleChangeCallback: jest.fn(),
    handleDeleteContentCallback: jest.fn(),
    handleSharedBoxChangeCallback: jest.fn(),
    isPrivate: true
  };
  const getShallowComponent = (props = defaultProps) => shallow(<JournalContent {...props} />);

  test('initailize states', () => {
    const component = getShallowComponent();
    expect(component.state('isPrivate')).toBeTruthy();
    expect(component.state(defaultProps.newContentKey)).toBe(defaultProps.newContent);
    expect(getShallowComponent({ ...defaultProps, isPrivate: false }).state('isPrivate')).toBeFalsy();
  });

  test('componentDidMount', () => {
    document.getElementsByTagName = jest.fn().mockReturnValue([{ value: 'value', style: {} }]);
    const tooltip = jest.fn();
    window.$ = jest.fn(() => ({ tooltip }));
    getShallowComponent();
    expect(window.$).toHaveBeenLastCalledWith('[data-toggle="tooltip"]');
    expect(tooltip).toHaveBeenCalledTimes(1);

    document.getElementsByTagName = jest.fn().mockReturnValue([{ value: '', style: {} }]);
    getShallowComponent();
  });

  test('handleChange', () => {
    const component = getShallowComponent();
    component.find('textarea').simulate('change', { target: { value: 'newValue' } });
    expect(component.state(defaultProps.newContentKey)).toBe('newValue');
    expect(defaultProps.handleChangeCallback).toHaveBeenLastCalledWith(defaultProps.newContentKey, 'newValue');
  });

  test('handleClose', () => {
    const component = getShallowComponent();
    component.find({ role: 'button' }).simulate('click');
    expect(defaultProps.handleDeleteContentCallback)
      .toHaveBeenLastCalledWith(defaultProps.newContentKey);
  });

  test('handleSharedBoxChange', () => {
    const component = getShallowComponent();
    const originalIsPrivate = component.state('isPrivate');
    component.find('input').simulate('change');
    const newIsPrivate = component.state('isPrivate');
    expect(newIsPrivate).toBe(!originalIsPrivate);
    expect(defaultProps.handleSharedBoxChangeCallback)
      .toHaveBeenLastCalledWith(defaultProps.newContentKey, newIsPrivate);
  });

  test('JournalContent snapshot', () => expect(renderer.create(<JournalContent {...defaultProps} />).toJSON()).toMatchSnapshot());
});
