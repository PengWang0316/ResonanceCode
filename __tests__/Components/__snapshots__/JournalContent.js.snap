// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalContent test JournalContent snapshot 1`] = `
<div
  className="form-group mt-3"
>
  <i
    className="fa fa-window-close closeIconSpan"
    onClick={[Function]}
    role="button"
    tabIndex="-1"
    title="Remove newContentName"
  />
  <div>
    <div
      className="row d-flex justify-content-between"
    >
      <div
        className="col-xs-6"
      >
        <label
          className="col-form-label text-capitalize mt-2 pl-4"
          htmlFor="newContentKey"
        >
          <b>
            newContentName
          </b>
        </label>
      </div>
      <div
        className="col-xs-6"
      >
        <label
          className="col-form-label checkboxInline"
          data-placement="top"
          data-toggle="tooltip"
          htmlFor="privateCheckbox"
          title="Checking this box can keep this content invisible after you share this journal to someone else."
        >
          <input
            checked={true}
            id="privateCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
           Private
        </label>
      </div>
    </div>
  </div>
  <div>
    <textarea
      className="form-control"
      id="newContentKey"
      onChange={[Function]}
      placeholder="newContentName..."
      rows="3"
      type="text"
      value="newContent"
    />
  </div>
</div>
`;
